### PR TITLE
CrayPE fix for libcircle

### PR DESCRIPTION
**Pull request by lukebroskop**
disable mpi search when using cce (the CCE compiler wrappers do not behave like the conventional mpi compiler wrappers)

### CODE REVIEW

File: var/spack/repos/builtin/packages/libcircle/package.py [hunk 1]
```diff
@@ -22,6 +22,12 @@ class Libcircle(AutotoolsPackage):
     depends_on('pkgconfig', type='build')
     depends_on('libpciaccess', type='link')
 
+    patch('CrayPE_configure-ac.patch', when='%cce')
+
+    @when('%cce')
+    def autoreconf(self, spec, prefix):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libcircle/package.py]: Have you considered using `force_autoreconf` as described with the second example at https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html?highlight=force_autoreconf#force-autoreconf?

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/libcircle/package.py]: In other words, something along the lines:

```
    @property
    def force_autoreconf(self):
        return self.spec.satisfies('%cce')
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
See suggestion

File: var/spack/repos/builtin/packages/libcircle/package.py (hunk 1 above)
Comment (lukebroskop) [reply var/spack/repos/builtin/packages/libcircle/package.py]: I like this approach better, thanks! I did have to add a couple additional dependencies (e.g. autoconf and automake).

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**